trunk/src/modules/upscale.F90

```diff
@@ -110,68 +160,97 @@ subroutine load_vis(V,Fp,Etype)
       end do
 !
       read(nin,*) V%NR_ELEM
-      allocate(V%ELEM(1:V%NR_ELEM, 0:8), STAT=istat)
+      allocate(V%ELEM(1:V%NR_ELEM, 0:27), STAT=istat)
```

Comment:
This change is only needed when using `SECOND_ORDER_VIS` ?

Reply:
Yes; first order vis only has a maximum of 8 vertices per element. For second order I stretched the definition of vertex to mean Lagrange points. We can redesign it a bit to be more clear if needed but then I think we either need to make separate infrastructure for second order or it will make first order less clear.

Reply:
why not just put
```suggestion
      if (SECOND_ORDER_VIS) then
         allocate(V%ELEM(1:V%NR_ELEM, 0:27), STAT=istat)
      else
         allocate(V%ELEM(1:V%NR_ELEM, 0:8), STAT=istat)
      endif
```

Reply:
I just figured things already weren't sized correctly for all first order element types but using 27 instead of 8 is a bit worse than using 8 instead of 4. This works!